### DISCUSSION TITLE
Insert.getItemsList deprecated and broken in 4.5

### DISCUSSION THREAD

**Discussion by jcamerin** [JSQLParser/JSqlParser#1602] [Q&A] [answered]
I recently tried to upgrade from 4.4 to 4.5 and one of my unitTests that relies on Insert.getItemsList began failing.  In 4.5, this method is marked as deprecated, but there is nothing I could find in the release notes related to this change/deprecation, and there is not any commentary in the source code WRT what replaces this functionality, i.e. what I should use instead of getItemsList.  
Does anyone know what the related change is and what I should be using instead of getItemsList?

**Comment by manticore-projects** [comment, marked as answer]
This one, SetOperations:

```
INSERT supports SetOperations (e. g. INSERT INTO ... SELECT ... FROM ... UNION SELECT ... FROM ...), those SetOperations are used both for SELECT and VALUES clauses (API change) in order to simplify the Grammar
```

**Reply by jcamerin** [reply]
Appreciate that info.  I was hoping to find some clues in the change as to why it is deprecated and what the preferred calls would be for the information provided by the now deprecated API on that class.  Maybe its just my lack of familiarity with the code, but it seems as thought it was just marked as deprecated without a replacement.  Am I missing something?

**Reply by manticore-projects** [reply]
Maybe the Syntaxt Diagrams will help: http://217.160.215.75:8080/jsqlformatter/_static/JSqlParserCC.xhtml#Insert
